spring-grpc-core/src/main/java/org/springframework/grpc/server/lifecycle/GrpcServerLifecycle.java

```diff
@@ -107,6 +105,11 @@ public int getPort() {
 		return this.server == null ? 0 : this.server.getPort();
 	}
 
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher eventPublisher) {
```

Comment:
I believe this is the more Spring idiomatic way.

Reply:
I don't think it matters does it? Don't we generally prefer constructor injection?

Reply:
I tend to see `ApplicationContext` and `AEP` passed in via the `<thing>Aware` contract. But, it is probably more of a preference. Constructor injection is a good thing too and make it easier to test. I will omit this from the proposal.

Reply:
Done